src/core/map/level_generator/generator_data.cpp

```diff
@@ -0,0 +1,311 @@
+#include "generator_data.h"
+#include "random.h"
+
+namespace map {
+
+GeneratorData::GeneratorData()
+{
+
+}
+
+GCell& GeneratorData::GetGCell( glm::vec2 pos )
+{
+    return mGCells[pos.y][pos.x];
+}
+
+GCell const& GeneratorData::GetGCell( glm::vec2 pos ) const
+{
+    return mGCells[pos.y][pos.x];
+}
+
+void GeneratorData::SetDimensions( int32_t x, int32_t y )
+{
+    mDimX = x;
+    mDimY = y;
+    mGCells.clear();
+    mGCells.resize( mDimY );
```

Comment:
fyi, resize can get a value_type arg, so the iteration with other resizes could be done in this same line, resize( mDimY, vector<Cell>( mDimX ) )